### DISCUSSION TITLE
lm_optimizer.py error: NameError: name 'lm_opt' is not defined

### DISCUSSION THREAD

**Discussion by FrontierDK** [coqui-ai/STT#2192] [Q&A]
Hi all.

When running the lm_optimizer.py, it does run and the 3 files are generated.

I Writing train set to /home/me/files/train.csv
I Writing dev set to /home/me/files/dev.csv
I Writing test set to /home/me/files/test.csv

But, it also finishes with an error. 
``` Traceback (most recent call last):
  File "/home/bruger/Python-3.7.6/STT/lm_optimizer.py", line 16, in <module>
    lm_optimize.main()
  File "/home/bruger/Python-3.7.6/STT/training/coqui_stt_training/util/lm_optimize.py", line 86, in main
    results = lm_opt.compute_lm_optimization()
NameError: name 'lm_opt' is not defined
```

**Comment by clenki** [comment]
Hey man,

I had this problem today aswell and needed about 2 hours to solve it. You have an old version of lm_optimize.py. 

Solution 1: Download the latest version of lm_optimize.py one here https://github.com/coqui-ai/STT/blob/main/training/coqui_stt_training/util/lm_optimize.py 
Solution 2: Simply edit line 86: Go into /home/bruger/Python-3.7.6/STT/training/coqui_stt_training/util/lm_optimize.py file and edit line 86 from
results = **lm_opt**.compute_lm_optimization()
to 
results = compute_lm_optimization()

After that it should start. Hope this helps.

**Reply by wasertech** [reply]
This is a [known bug](https://github.com/coqui-ai/STT/issues/2202). Just update to STT 1.4.0 that contains [my fixes for it](https://github.com/coqui-ai/STT/pull/2203). No need to mangle with code 😉
